src/api/resources/mixins.py

```diff
@@ -260,6 +264,31 @@ def edit(self, request, project_pk, pk):
             return Response(str(err), status=500)
 
 
+class SampleUnitMethodSummaryReport(object):
+    @action(
+        detail=False, methods=["GET"], permission_classes=[ProjectDataAdminPermission]
+    )
+    def xlsx(self, request, project_pk):
+        from ..utils.reports import create_sample_unit_method_summary_report
+        try:
+            model = self.get_queryset().model
+            with NamedTemporaryFile(delete=False) as f:
+                create_sample_unit_method_summary_report(
+                    project_pk,
+                    getattr(model, "protocol"),
+                    f.name,
+                    request=request
+                )
+                response = HttpResponse(content_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
+                response["Content-Length"] = os.fstat(f.fileno()).st_size
+                response["Content-Disposition"] = f'attachment; filename="summary_report_{project_pk}.xlsx"'
```

Comment:
I think filename should be something more liike
`<sanitized_project_name>_<protocol>_<iso_date>.xlsx`
(I feel like we have a utility somewhere for producing  `sanitized_project_name`)